internal/webhooks/inclusterippool.go

```diff
@@ -212,6 +212,21 @@ func (webhook *InClusterIPPool) validate(_, newPool types.GenericInClusterPool)
 		}
 	}
 
+	for _, address := range newPool.PoolSpec().ExcludedAddresses {
+		ipSet, err := poolutil.AddressToIPSet(address)
+		if err != nil {
+			allErrs = append(allErrs, field.Invalid(field.NewPath("spec", "excludedAddresses"), address, "provided address is not a valid IP, range, nor CIDR"))
+			continue
+		}
+		from := ipSet.Ranges()[0].From()
+		hasIPv4Addr = hasIPv4Addr || from.Is4()
+		hasIPv6Addr = hasIPv6Addr || from.Is6()
+	}
+
+	if hasIPv4Addr && hasIPv6Addr {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("spec", "excludedAddresses"), newPool.PoolSpec().ExcludedAddresses, "provided addresses are of mixed IP families"))
```

Comment:
We need to make sure that the family of `excludedAddresses` matches the family of `addresses`, and not that the family is the same for all. This approach _should_ work, but it would be good to be a bit more explicit about it.

Reply:
No problem, will add an additional check. 

I initially thought that it wouldn't really matter as we are creating a set of IPs from the provided addresses and then just removing whatever is being found in the excluded addresses. Therefore if you attempt to remove ipv6 addresses from ipv4 addresses and vice versa, nothing will happen.